### PR TITLE
Soumyo/aws ec2 transit gateway route table association

### PR DESCRIPTION
**Pull request by soumyo13**
### Description

`aws ec2 transit gateway route table association` InSpec audit resource to test properties of a ec2 transit gateway route table association.

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: test/unit/resources/aws_ec2_transit_gateway_route_table_association_test.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# require 'helper'
```
Comment (sathish-progress) [inline test/unit/resources/aws_ec2_transit_gateway_route_table_association_test.rb:1]: tests are empty!!

File: libraries/aws_ec2_transit_gateway_route_table_associations.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  end
+
+  def fetch_data
+    rows = []
+    @query_params[:max_results] = 1000
+    loop do
+      catch_aws_errors do
+        @api_response = @aws.compute_client.get_transit_gateway_route_table_associations(@query_params)
+      end
+      return [] if !@api_response || @api_response.empty?
```
Comment (sathish-progress) [inline libraries/aws_ec2_transit_gateway_route_table_associations.rb:40]: return rows here

**Comment by sathish-progress** [comment]
add docs link to readme

File: libraries/aws_ec2_transit_gateway_route_table_associations.rb (hunk 2 above)
Comment (soumyo13) [reply libraries/aws_ec2_transit_gateway_route_table_associations.rb:40]: done

File: docs/resources/aws_ec2_transit_gateway_route_table_association.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    describe aws_ec2_transit_gateway_route_table_association(transit_gateway_route_table_id: 'TRANSIT_GATEWAY_ROUTE_TABLE_ID') do
+      it { should exist }
+    end
+
+## Parameters
+
+`transit_gateway_route_table_id` _(required)_
+
+For additional information, see the [AWS documentation on transit gateway route table association](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-transitgatewayroutetableassociation.html).
```
Comment (dishanktiwari2501) [inline docs/resources/aws_ec2_transit_gateway_route_table_association.md:24]: For additional information, check out the [AWS documentation on transit gateway route table association](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-transitgatewayroutetableassociation.html).<br class="Apple-interchange-newline">

File: docs/resources/aws_ec2_transit_gateway_route_table_association.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+### Ensure that the state is available or deleted.
+
+    describe aws_ec2_transit_gateway_route_table_association(transit_gateway_route_table_id: 'TRANSIT_GATEWAY_ROUTE_TABLE_ID') do
+        its('state') { should eq 'available' }
+    end
+
+## Matchers
+
+This InSpec audit resource has the following special matchers. For a full list of available matchers, please visit our [Universal Matchers page](https://www.inspec.io/docs/reference/matchers/).
```
Comment (dishanktiwari2501) [inline docs/resources/aws_ec2_transit_gateway_route_table_association.md:51]: This InSpec audit resource has the following special matchers. For the complete list of available matchers, visit [Universal Matchers page](https://www.inspec.io/docs/reference/matchers/).<br class="Apple-interchange-newline">

File: docs/resources/aws_ec2_transit_gateway_route_table_association.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+The controls will pass if the `describe` method returns at least one result.
+
+### exist
+
+Use `should` to test that the entity exists.
```
Comment (dishanktiwari2501) [inline docs/resources/aws_ec2_transit_gateway_route_table_association.md:57]: Use `should` to test whether the entity exists.<br class="Apple-interchange-newline">

File: docs/resources/aws_ec2_transit_gateway_route_table_association.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+    describe aws_ec2_transit_gateway_route_table_association(transit_gateway_route_table_id: 'TRANSIT_GATEWAY_ROUTE_TABLE_ID') do
+      it { should be_available }
+    end
+
+## AWS Permissions
+
+Your [Principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/intro-structure.html#intro-structure-principal) will need the `ec2:client:get_transit_gateway_route_table_associations` action with `Effect` set to `Allow`.
+
+See the [Actions, Resources, and Condition Keys for Amazon EC2](https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazonec2.html) documentation for additional information.
```
Comment (dishanktiwari2501) [inline docs/resources/aws_ec2_transit_gateway_route_table_association.md:81]: For additional information, check out the [Actions, Resources, and Condition Keys for Amazon EC2](https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazonec2.html) documentation.<br class="Apple-interchange-newline">

File: docs/resources/aws_ec2_transit_gateway_route_table_associations.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+    describe aws_ec2_transit_gateway_route_table_associations(transit_gateway_route_table_id: 'TRANSIT_GATEWAY_ROUTE_TABLE_ID') do
+      it { should exist }
+    end
+
+## Parameters
+
+`transit_gateway_route_table_id` _(required)_
+
+See the [AWS documentation on transit gateway route table](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-transitgatewayroutetableassociation.html) for additional information.
```
Comment (dishanktiwari2501) [inline docs/resources/aws_ec2_transit_gateway_route_table_associations.md:24]: For additional information, check out the [AWS documentation on transit gateway route table](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-transitgatewayroutetableassociation.html) documentation.<br class="Apple-interchange-newline">

File: docs/resources/aws_ec2_transit_gateway_route_table_associations.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+| Property | Description|
+| --- | --- |
+| transit_gateway_route_table_ids | The ID of the transit gateway route table. |
+| resource_types | The resource type. Valid values are: `vpc`, `vpn`, `direct-connect-gateway`, `peering`, and `connect`. |
+| resource_ids | The ID of the resource. |
+| states | The possible states of the route table are: `available`, `deleting`, `deleted`, and `pending`. |
+
+## Examples
+
+##### Ensure a transit gateway route table exists.
```
Comment (dishanktiwari2501) [inline docs/resources/aws_ec2_transit_gateway_route_table_associations.md:37]: ##### Ensure a transit gateway route table exists<br class="Apple-interchange-newline">

File: docs/resources/aws_ec2_transit_gateway_route_table_associations.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+    describe aws_ec2_transit_gateway_route_table_associations(transit_gateway_route_table_id: 'TRANSIT_GATEWAY_ROUTE_TABLE_ID') do
+      it { should exist }
+    end
+
+##### Match count of transit gateway route table.
```
Comment (dishanktiwari2501) [inline docs/resources/aws_ec2_transit_gateway_route_table_associations.md:43]: ##### Match count of transit gateway route table<br class="Apple-interchange-newline">

File: docs/resources/aws_ec2_transit_gateway_route_table_associations.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+    describe aws_ec2_transit_gateway_route_table_associations(transit_gateway_route_table_id: 'TRANSIT_GATEWAY_ROUTE_TABLE_ID') do
+        its('count') { should eq 5 }
+    end
+
+##### Check the state of the route table.
```
Comment (dishanktiwari2501) [inline docs/resources/aws_ec2_transit_gateway_route_table_associations.md:49]: ##### Check the state of the route table<br class="Apple-interchange-newline">

File: docs/resources/aws_ec2_transit_gateway_route_table_associations.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+       its('states') { should include "available" }
+    end
+
+## Matchers
+
+This InSpec audit resource has the following special matchers. For a full list of available matchers, please visit our [Universal Matchers page](https://www.inspec.io/docs/reference/matchers/).
+
+#### exist
+
+The control will pass if the describe returns at least one result.
```
Comment (dishanktiwari2501) [inline docs/resources/aws_ec2_transit_gateway_route_table_associations.md:61]: The control will pass if the `describe` returns at least one result.<br class="Apple-interchange-newline">

File: docs/resources/aws_ec2_transit_gateway_route_table_associations.md [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+
+Use `should` to test the entity should exist.
```
Comment (dishanktiwari2501) [inline docs/resources/aws_ec2_transit_gateway_route_table_associations.md:63]: Use `should` to test whether the entity should exist.<br class="Apple-interchange-newline">

File: docs/resources/aws_ec2_transit_gateway_route_table_associations.md [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+
+    describe aws_ec2_transit_gateway_route_table_associations(transit_gateway_route_table_id: 'TRANSIT_GATEWAY_ROUTE_TABLE_ID') do
+      it { should_not exist }
+    end
+
+## AWS Permissions
+
+Your [Principal](https://docs.aws.amazon.com/IAM/latest/UserGuide/intro-structure.html#intro-structure-principal) will need the `ec2:client:get_transit_gateway_route_table_associations` action with Effect set to Allow.
+
+You can find detailed documentation at [Actions, Resources, and Condition Keys for Amazon EC2](https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazonec2.html).
```
Comment (dishanktiwari2501) [inline docs/resources/aws_ec2_transit_gateway_route_table_associations.md:79]: You can find the detailed documentation at [Actions, Resources, and Condition Keys for Amazon EC2](https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazonec2.html).<br class="Apple-interchange-newline">

**Review by dishanktiwari2501 — APPROVED** [review]

**Review by Nirbhay1997 — APPROVED** [review]

**Pull request merged**